webapp/src/components/root/root.jsx

```diff
@@ -82,40 +93,90 @@ export default class Root extends React.Component {
                 >
                     <h1>{'Add a Todo'}</h1>
                     <div className='todoplugin-issue'>
-                        <h2>
-                            {'Todo Message'}
-                        </h2>
-                        <textarea
-                            className='todoplugin-input'
-                            style={style.textarea}
-                            value={message}
-                            onChange={(e) => this.setState({message: e.target.value})}
-                        />
+                        <h2>{'Todo Message'}</h2>
+                        <div className={'btn-group'}>
+                            <button
+                                className={`btn ${
+                                    !this.state.previewMarkdown && 'btn-primary'
+                                }`}
+                                onClick={() => {
+                                    this.setState({previewMarkdown: false});
+                                }}
+                            >
+                                {'Write'}
+                            </button>
+                            <button
+                                className={`btn ${
+                                    this.state.previewMarkdown && 'btn-primary'
+                                }`}
+                                onClick={() => {
+                                    this.setState({previewMarkdown: true});
+                                }}
+                            >
+                                {'Preview'}
+                            </button>
+                        </div>
+                        {this.state.previewMarkdown ? (
+                            <div
+                                className='todoplugin-input'
+                                style={style.markdown}
+                            >
+                                {PostUtils.messageHtmlToComponent(
+                                    PostUtils.formatText(this.state.message),
+                                )}
+                            </div>
+                        ) : (
+                            <textarea
+                                className='todoplugin-input'
+                                style={style.textarea}
+                                value={message}
+                                onChange={(e) =>
+                                    this.setState({message: e.target.value})
+                                }
+                            />
+                        )}{' '}
```

Comment:
I think this looks better with space on its own line. Also
```suggestion
                        )}
                        {' '}
```